designs/cloudWatch/cloudWatchIntegration.md

```diff
@@ -0,0 +1,76 @@
+
+## Comparison to other products
+
+| Product                       | Difference    |
+|-------------------------------|---------------|
+| AWS Toolkit for Visual Studio | Not supported |
+| AWS Toolkit for VSCode        | Not supported |
+
+## Experience
+
+### View Log Groups
+A new AWS Explorer root node would be added for the CloudWatch service. Underneath of that node we would add the list of Log Groups that exist in the selected region for the selected credential profile. Double clicking a Log Group would open the CloudWatch tool window with the tab for the Log Streams that belong to that group. A new tab would be created if needed.
+
+*Browsing Log Groups in AWS Explorer:*<br />
+![BROWSE_LOG_GROUP]
+
+#### Alternative Proposal
+Instead of building a new CloudWatch service node in the AWS Explorer, we instead build a CloudWatch Log Group browser tab in the CloudWatch tool window. This new tab would also be linked to the active AWS connection settings. This would differ from the other tabs which may lead to a confusing user experience and seen as a con.
+
+*Possible alternative for browsing Log Groups:*<br />
+![ALT_BROWSE_LOG_GROUP]
+
+### View Log Streams
+Upon selecting a Log Group, the user would be shown a new tab in the CloudWatch tool window. This new tab will list all the streams for the requested group to allow the user to find the exact stream they are looking for.
+
+Log Streams can also be sorted by name or last event time in order to help facilitate finding the Log Stream of interest.
+
+*View all Log Streams in the selected Log Group:*<br />
+![BROWSE_LOG_STREAMS]
+
+#### Navigate to a Log Stream
+Double clicking the Log group in the AWS Explorer will open the Log Stream selection. 
+
+For additional service integrations, such as a Lambda function, where we know what Log Group is associated with the resource a new context menu action adds the ability to bypass having to search for the Log Group and instead open the Log Stream selection directly.
+
+### View A Log Stream
+Each Log Stream the user selects opens in its own tab to allow for browsing multiple streams at the time. The tab is also be scoped to the region and credentials used to open it so that the user can browse across regions and accounts at the same time.
+
+*View of a Log Stream:*<br />
+![VIEW_LOG_STREAM]
+
+#### Treat as...
+The View As feature acts as a Log Event post-processor to convert the Log Event into a more readable format. This feature provides a pre-defined list of formats such as JSON, Plain Text, CSV, or [CLF](https://httpd.apache.org/docs/1.3/logs.html#common). The formats would process each Log Event and convert them into a more tabular format where each column maps to an extract format. The filter feature would filter on these extracted values.
+
+*Treat As -> JSON:*<br />
```

Comment:
If one field contains a json blob itself, do you take further action, or treat it as a string?

Reply:
Treat it as a string, the entire log line would need to be JSON to be trated correctly.